### DISCUSSION TITLE
Drag 'n Swap

### DISCUSSION THREAD

**Discussion by mgutz** [i3/i3#5422] [Ideas]
The Drag 'n Drop by mouse is wonderful. I currently use i3 as the WM for XFCE DE.  This combination is better than Gnome + tiling extensions (Pop or Forge) and the very limited tiling built into the latest KDE Plasma.

One feature I really miss form Awesome WM is the ability to swap windows by dropping a window over another. Currently, if I drag a window into the center of a window, i3 puts the dragged into that position. I found it less useful than dropping on an edge or gap. What if instead  the dragged window is swapped when dropped on the center?

**Comment by joedefen** [comment]
Copy that to this suggestion.  It is also a part of Sway, too.  I could live with it being optional behavior (although the current behavior is rather useless ... dropping anywhere but the center is pretty much my rule).  Or allow a python script, one way or another (e.g., it would be nice to know there was a drop-to-center and the window info to "make it right" somehow).

**Comment by orestisfl** [comment]
See https://github.com/i3/i3/pull/6084
